File: python/src/conversor.py
from modelos import Conversion, ConversionRegistro


BASES = [2, 8, 16]


def convertir_todas(numero_decimal: int) -> list[ConversionRegistro]:
    conversion = Conversion(numero_decimal)
    return [conversion.convertir(base) for base in BASES]


def crear_matriz_resultados(
    conversiones: list[ConversionRegistro],
) -> list[list[str]]:
    return [
        [str(conversion.base), conversion.resultado()]
        for conversion in conversiones
    ]

File: typescript/src/conversor.ts
import { Conversion, ConversionRegistro } from "./modelos";

const BASES = [2, 8, 16];

export function convertirTodas(numeroDecimal: number): ConversionRegistro[] {
  const conversion = new Conversion(numeroDecimal);
  return BASES.map((base) => conversion.convertir(base));
}

export function crearMatrizResultados(
  conversiones: ConversionRegistro[],
): string[][] {
  return conversiones.map((conversion) => [
    conversion.base.toString(),
    conversion.digitos.join(""),
  ]);
}
